src/components/MailboxPicker.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+	<Treeselect
+		ref="Treeselect"
+		:options="mailboxes"
+		:multiple="false"
+		:clearable="false"
+		v-bind="$attrs"
+		v-on="$listeners" />
+</template>
+<script>
+import Treeselect from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+
+export default {
+	name: 'MailboxPicker',
+	components: {
+		Treeselect,
+	},
+	props: {
+		accountid: {
+			type: Number,
+			required: true,
+		},
+	},
+	computed: {
+		mailboxes() {
+			const boxes = []
+			this.getMailboxes(boxes)
+			return boxes
+		},
+	},
+	methods: {
+		getMailboxes(root, folderid) {
```

Comment:
the function should take one optional argument `maliboxId` and return the result as array

Reply:
see below